database/gormdb_course.go

```diff
@@ -53,30 +54,50 @@ func (db *GormDB) CreateCourse(courseCreatorID uint64, course *qf.Course) error
 	return tx.Commit().Error
 }
 
-// GetCourse fetches course by ID. If withInfo is true, preloads course
-// assignments, active enrollments and groups.
-func (db *GormDB) GetCourse(courseID uint64, withEnrollments bool) (*qf.Course, error) {
-	m := db.conn
+// GetCourse fetches course by ID.
+func (db *GormDB) GetCourse(courseID uint64) (*qf.Course, error) {
 	var course qf.Course
+	if err := db.conn.First(&course, courseID).Error; err != nil {
+		return nil, err
+	}
+	return &course, nil
+}
 
-	if withEnrollments {
-		// we only want submission from users enrolled in the course
+func (db *GormDB) GetCourseByStatus(courseID uint64, status qf.Enrollment_UserStatus) (*qf.Course, error) {
+	m := db.conn
+	var course qf.Course
+	switch status {
```

Comment:
This method needs tests that cover the different status inputs. See the coverage complaints below.